src/ember-app.js

```diff
@@ -235,7 +239,7 @@ class EmberApp {
       })
       .then(() => result.instanceBooted = true)
       .then(() => instance.visit(path, bootOptions))
-      .then(() => waitForApp(instance))
+      .then(() => fastbootInfo.deferredPromise)
```

Comment:
I am not worried about the registry. I am talking about the mutations during `visit` which is on the instance and not the application. If you can verify the objects are same via a simple test of changing the request property on the `fastbootInfo` in your app, this should be good.